conu/backend/nspawn/container.py

```diff
@@ -0,0 +1,261 @@
+
+    def __repr__(self):
+        return "NspawnContainer(image=%s, id=%s)" % (self.image, self.get_id())
+
+    def __str__(self):
+        return self.get_id()
+
+    def get_id(self):
+        return self._id
+
+    def inspect(self, refresh=True):
+        """
+        return cached metadata by default (a convenience method)
+
+        :param refresh: bool, returns up to date metadata if set to True
+        :return: dict
+        """
+        return self.get_metadata(refresh=refresh)
+
+    def get_metadata(self, refresh=True):
+        """
+        return cached metadata by default
+
+        :param refresh: bool, returns up to date metadata if set to True
+        :return: dict
+        """
+        if refresh or not self._metadata:
+            ident = self._id or self.name
+            if not ident:
+                raise ConuException("This container does not have a valid identifier.")
+            self._metadata = output_decode(run_cmd(["machinectl", "show",  ident], return_output=True))
+        return self._metadata
+
+    def is_running(self):
+        return "running" == self.get_metadata(refresh=True)["State"]
+
+    def copy_to(self, src, dest):
+        """
+        copy a file or a directory from host system to a container
+
+        :param src: str, path to a file or a directory on host system
+        :param dest: str, path to a file or a directory within container
+        :return: None
+        """
+        # using `docker cp` b/c put_archive is too complicated
+        logger.debug("copying %s from host to container at %s", src, dest)
+        cmd = ["machinectl", "copy-to",self.get_id(), src, dest]
+        run_cmd(cmd)
+
+    def copy_from(self, src, dest):
+        """
+        copy a file or a directory from container or image to host system.
+
+        :param src: str, path to a file or a directory within container or image
+        :param dest: str, path to a file or a directory on host system
+        :return: None
+        """
+        # using `docker cp` b/c get_archive is too complicated
+        logger.debug("copying %s from host to container at %s", src, dest)
+        cmd = ["machinectl", "copy-from", self.get_id(), src, dest]
+        run_cmd(cmd)
+
+
+    def stop(self):
+        """
+        stop this container
+
+        :return: None
+        """
+        self.kill()
+
+    def kill(self, signal=None):
+        try:
+            run_cmd(["machinectl", "poweroff", self.get_id()])
+        except BaseException as poweroffex:
+            logger.debug("Unable to stop machine via poweroff, terminating : %s" % poweroffex)
+            try:
+                run_cmd(["machinectl", "terminate", self.get_id()])
+                self.image._is_killed(self.get_id())
+            except BaseException as poweroffexterm:
+                logger.debug("Unable to stop machine via terminate, STRANGE: %s" % poweroffexterm)
+                time.sleep(constants.DEFAULT_RETRYTIMEOUT)
```

Comment:
Please provide an explanation of this sleep. Could its absence affect methods called after?